tests/conftest.py

```diff
@@ -10,3 +19,141 @@
+    decoder.decode.return_value = VALID_DECODED_LIST
+
+    return decoder
+
+
+@pytest.fixture
+def invalid_decoder():
+    decoder = MagicMock()
+    decoder.decode.return_value = INVALID_DECODED_LIST
+
+    return decoder
+
+
+@pytest.fixture
+def valid_create_labware_message(unchecked_create_labware_message, valid_decoder):
+    unchecked_create_labware_message.decode(valid_decoder)
+
+    return unchecked_create_labware_message
+
+
+@pytest.fixture
+def invalid_create_labware_message(unchecked_create_labware_message, invalid_decoder):
+    unchecked_create_labware_message.decode(invalid_decoder)
+
+    return unchecked_create_labware_message
+
+
+@pytest.fixture
+def traction_success_creation_response():
+    return {
+        "data": {
+            "id": "52",
+            "type": "receptions",
+            "links": {"self": "http://localhost:3000/v1/receptions/52"},
+            "attributes": {"source": "traction-ui.sequencescape"},
+        }
+    }
+
+
+@pytest.fixture
+def taxonomy_record():
+    return {
+        "taxId": "9600",
+        "scientificName": "Pongo pygmaeus",
+        "commonName": "Bornean orangutan",
+        "formalName": "true",
+        "rank": "species",
+        "division": "MAM",
+        "lineage": (
+            "Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi; "
+            "Mammalia; Eutheria; Euarchontoglires; Primates; Haplorrhini; "
+            "Catarrhini; Hominidae; Pongo; "
+        ),
+        "geneticCode": "1",
+        "mitochondrialGeneticCode": "2",
+        "submittable": "true",
+        "binomial": "true",
+    }
+
+
+@pytest.fixture
+def valid_sample():
+    return {
+        "sampleUuid": "dd490ee5-fd1d-456d-99fd-eb9d3861e0f6".encode(),
+        "studyUuid": "dd490ee5-fd1d-456d-99fd-eb9d3861e014".encode(),
+        "sangerSampleId": "cichlid_pacbio8196429",
+        "location": "A01",
+        "supplierSampleName": "SampleSupplied1",
+        "volume": "5",
+        "concentration": "5",
+        "publicName": "SamplePublicName1",
+        "taxonId": "10090",
+        "commonName": "Mus Musculus",
+        "donorId": "cichlid_pacbio8196429",
+        "libraryType": "Library1",
+        "countryOfOrigin": "United Kingdom",
+        "sampleCollectionDateUtc": datetime.now(),
+    }
+
+
+@pytest.fixture
+def invalid_sample():
+    return {
+        "sampleUuid": "dd490ee5-fd1d-456d-99fd-eb9d3861e0f6",
+        "studyUuid": "dd490ee5-fd1d-456d-99fd-eb9d3861e014",
+        "sangerSampleId": 1234,
+        "location": "A001",
+        "supplierSampleName": "1234",
+        "volume": "ee5",
+        "concentration": "ee5",
+        "publicName": "1234",
+        "taxonId": "ee10090",
+        "commonName": 1234,
+        "donorId": 1234,
+        "libraryType": 1234,
+        "countryOfOrigin": 1234,
+        "sampleCollectionDateUtc": datetime.now(),
+    }
+
+
+def read_file(filename):
+    file = open(filename, "r")
+    contents = ""
+
+    while True:
```

Comment:
Why do we need a loop here instead of a single file.read() call?

Reply:
with open(filename) as file:
    return file.read()

Reply:
yes, I'll change it